src/analyzers/type-refactoring/du-incremental/plan-generator.ts

```
@@ -0,0 +1,869 @@
+ * Helper Functions for ${typeGeneration.unionType.typeName}
+ * Generated by funcqc DU transformation
+ */
+
+import type { ${typeGeneration.unionType.typeName} } from '../types/${typeGeneration.unionType.typeName.toLowerCase()}.types';
+
+`;
+
+    // Add smart constructors
+    if (typeGeneration.smartConstructors.length > 0) {
+      content += '// Smart Constructors\n';
+      typeGeneration.smartConstructors.forEach(constructor => {
```

Biome (2.1.2)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)